### PR TITLE
Minor improvements to harden TF code

### PR DESCRIPTION
**Pull request by sandrine-bailleux-arm**
This patch stack introduces minor improvements to harden the code:
some extra checks on input values, some extra debug assertions.

### CODE REVIEW

File: bl31/runtime_svc.c [hunk 1]
```diff
@@ -80,33 +84,32 @@ static int32_t validate_rt_svc_desc(rt_svc_desc_t *desc)
  ******************************************************************************/
 void runtime_svc_init(void)
 {
-	int32_t rc = 0;
-	uint32_t index, start_idx, end_idx;
-	uint64_t rt_svc_descs_num;
+	int rc = 0, index, start_idx, end_idx;
```
Comment (sorenb-xlnx) [inline bl31/runtime_svc.c]: As the types are reworded, should these indices probably become *size_t*?

Comment (soby-mathew) [inline bl31/runtime_svc.c]: This change was primarily targeted towards variables that need to be of different word size depending on the Architecture(AArch32/AArch64) it is compiled for. Typically variables holding address (uintptr_t), address range (size_t) were addressed as part of the type rework.

The rest of generic variables were not changed. The `indices` here come under that category.

**Comment by danh-arm** [comment]
Needs rebasing

**Comment by sandrine-bailleux-arm** [comment]
Rebased.

**Pull request merged**